src/firmware/shared/circular_buffer.h

```diff
@@ -0,0 +1,46 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/**
+ * This struct represents the circular buffer properties
+ */
+typedef struct CircularBuffer CircularBuffer_t;
+
+/**
+ * Initialize the circular_buffer struct members
+ * @param  size Size of the circular_buffer to set
+ *
+ * @return A pointer to the created circular_buffer, ownership is given to the caller
+ */
+CircularBuffer_t *circular_buffer_create(size_t size);
+
+/**
+ * Push data into the circular_buffer
+ * If full, will overwrite the oldest value
+ * @param  cbuffer The circular_buffer
+ * @param  data  Value to be inserted into the circular_buffer
+ */
+void circular_buffer_push(CircularBuffer_t *cbuffer, float data);
+
+/**
+ * Retrieve the most recent value in the circular_buffer
+ * The index is relative to the most recent value in the buffer.
+ * Eg. 1 is the most recent value, 2 is the second most recent, etc
```

Comment:
Comment should specify 0 is invalid input.

On another note, what do people think about making `0` access the most recent value, and then go from there? I think that would be reasonable enough since indices starting at 0 is common in C (so shouldn't be super confusing to anyone), and then we don't have just a single invalid value.

@kelvinkoon @EvanMorcom @garethellis0 @akhilveeraghanta @jonathanlew

Reply:
Made `circular_buffer_getAtIndex` start at 0 and updated comment accordingly.
In `circular_buffer_getAtIndex`, is there any need to do the `assert(index > -1)` to check for negatives since uint8_t is unsigned?

Reply:
> is there any need to do the assert(index > -1) to check for negatives since uint8_t is unsigned?

no that's no longer needed